data/colors/mixins/dark_mode.scss

```diff
@@ -360,44 +360,44 @@ $export: (
   ),
 
   toast: (
-    text: $blue-2,
-    bg: $black,
-    border: #214981,
+    text: var(--color-text-primary),
+    bg: $gray-6,
+    border: $gray-4,
     shadow: var(--color-shadow-large),
-    icon: $blue-2,
-    icon-bg: #121D2E,
-    icon-border: #214981,
+    icon: $white,
+    icon-bg: $blue-5,
+    icon-border: $blue-4,
 
     success: (
-      text: $green-2,
-      border: #1C532C,
-      icon: $green-2,
-      icon-bg: #111F1C,
-      icon-border: #1C532C,
+      text: var(--color-text-primary),
+      border: $gray-4,
+      icon: $white,
+      icon-bg: $green-4,
+      icon-border: $green-3,
     ),
 
     warning: (
-      text: $yellow-2,
-      border: #5E4411,
-      icon: $yellow-2,
-      icon-bg: #1F1C16,
-      icon-border: #5E4411,
+      text: var(--color-text-primary),
+      border: $gray-4,
+      icon: $white,
+      icon-bg: $yellow-4,
+      icon-border: $yellow-3,
     ),
 
     danger: (
-      text: $red-3,
-      border: #792E2E,
-      icon: $red-3,
-      icon-bg: #25171C,
-      icon-border: #792E2E,
+      text: var(--color-text-primary),
+      border: $gray-4,
+      icon: $white,
+      icon-bg: $red-5,
+      icon-border: $red-4,
     ),
 
     loading: (
       text: var(--color-text-primary),
-      border: var(--color-border-primary),
+      border: $gray-4,
       icon: $white,
-      icon-bg: $gray-6,
-      icon-border: var(--color-border-primary),
+      icon-bg: $gray-4,
+      icon-border: $gray-3,
```

Comment:
@auareyou The `icon-border` is not defined in the spec here https://github.com/github/primer/issues/57#issuecomment-804044793, so I just used `$gray-3` (one level higher than `icon-bg`).